moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp

```diff
@@ -827,6 +827,30 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
     }
   }
 
+  MoveItErrorCode plan(Plan& plan_, const std::string& pipeline_id, const std::string& planner_id)
+  {
+    const std::string current_pipeline_id = planning_pipeline_id_;
+    const std::string current_planner_id = planner_id_;
+    setPlanningPipelineId(pipeline_id);
+    setPlannerId(planner_id);
+    MoveItErrorCode error_code = plan(plan_);
+    setPlanningPipelineId(current_pipeline_id);
+    setPlannerId(current_planner_id);
+    return error_code;
+  }
+
+  MoveItErrorCode move(bool wait, const std::string& pipeline_id, const std::string& planner_id)
+  {
+    const std::string current_pipeline_id = planning_pipeline_id_;
+    const std::string current_planner_id = planner_id_;
+    setPlanningPipelineId(pipeline_id);
+    setPlannerId(planner_id);
+    MoveItErrorCode error_code = move(wait);
+    setPlanningPipelineId(current_pipeline_id);
+    setPlannerId(current_planner_id);
+    return error_code;
+  }
```

Comment:
This is unnecessarily thread-unsafe and the alternative is not any more complicated (as you seem to imply).

Just modify `MoveGroupInterfaceImpl::move` and `::plan` to take the additional parameters you need.
If you plan to add any more overrides in these calls in the future, maybe make [use an options struct](https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments).
Defining **5** new wrapper methods when the MGI already uses pimpl does not make sense.

Reply:
Right? What was I thinking. Christ.